specifications/diagrams/state-diagrams/payment-proposal-initiator.txt

```diff
@@ -0,0 +1,20 @@
+stateDiagram-v2
+    [*] --> Idle
+    note left of [*]: Initiator proposing a payment
+
+    Idle --> Connecting: Connect()
+    Connecting --> Listening: handshake successful
+    Connecting --> Idle: handshake failed
+    
+    %% Proposing:
+    Listening --> Proposing: ProposePayment() <br>Create D_i, C_i <br> Sign C_i <br> Send both to R
+    Proposing --> ResponseReceived
+    note right of Proposing: waiting/listening while in proposing state
+    Proposing --> NoResponseReceived: timeout expires
+
+    NoResponseReceived --> Listening
+    ResponseReceived --> Listening: Response invalid, or rejected
+    ResponseReceived --> ProposalRejected: rejected, timeout error, <br> invalid response
+    ProposalRejected --> Listening
+    ResponseReceived --> ProposalAccepted: valid Response <br> receieved signed C_i/D_i
+    ProposalAccepted --> Listening: sign D_i and send both back
```

Comment:
I think this is a factor of this being a state diagram and a flow chart in one, that this doesn't capture what happens if the initiator doesn't return the final signatures to the responder.